test/object-store/sync/flx_sync.cpp

```diff
@@ -4602,6 +4602,112 @@ TEST_CASE("flx sync: Client reset during async open", "[sync][flx][client reset]
     REQUIRE(subscription_invoked.load());
 }
 
+// Test for issue #6839, where async wait for download then async wait for subscription state after
+// commiting a new subscription was leading to a garbage reason value in the status provided to the
+// subscription complete callback.
+TEST_CASE("flx sync: no garbage data if wait for download prior to wait subs complete", "[sync][flx][baas]") {
```

Comment:
this test almost fully duplicates "flx: query on non-queryable field results in query error message" which was failing also with this error on jenkins. 

Could you merge them with these additional checks for wait_for_download and sync/async status checks?

It's just not like this test reliably crashes (as the other one), even under asan.

Reply:
I merged the new test with the existing test. I also tested it with the `subscriptions.cpp` change reverted to make sure it would still catch the original error.